protocol/messenger_contacts.go

```diff
@@ -16,6 +16,55 @@ import (
 	"github.com/status-im/status-go/protocol/transport"
 )
 
+func (m *Messenger) prepareMutualStateUpdateMessage(contactID string, updateType MutualStateUpdateType, clock uint64, timestamp uint64, outgoing bool) (*common.Message, error) {
+	var text string
+
+	var to string
+	var from string
+	if outgoing {
+		to = contactID
+		from = m.myHexIdentity()
+
+		switch updateType {
+		case MutualStateUpdateTypeSent:
+			text = "You sent a contact request to @" + to
+		case MutualStateUpdateTypeAdded:
+			text = "You added  @" + to + " as a contact"
+		case MutualStateUpdateTypeRemoved:
+			text = "You removed @" + to + " as a contact"
+		}
+	} else {
+		to = m.myHexIdentity()
+		from = contactID
+
+		switch updateType {
+		case MutualStateUpdateTypeSent:
+			text = "@" + from + " sent you a contact request"
+		case MutualStateUpdateTypeAdded:
+			text = "@" + from + " added you as a contact"
+		case MutualStateUpdateTypeRemoved:
```

Comment:
Since Go doesn't have pattern matching like Rust which does exhaustive checks, if a new `MutualStateUpdateType` is added, the compiler won't complain. This is by design, I'm not saying Rust is better. I'm wondering if it wouldn't be more robust to generate an error if no match is found, otherwise `text` will be empty, or at least have a default (non-empty) `text`. Maybe this is not the best example, since it's not a big deal. Anyways, wdyt?

Reply:
I like the rust approach better too. Fixed :)